doc/user/client.md

```diff
@@ -0,0 +1,443 @@
+# Operator-SDK: Controller Runtime Client API
+
+## Overview
+
+The [`controller-runtime`][repo-controller-runtime] library provides various abstractions to watch and reconcile resources in a Kubernetes cluster via CRUD (Create, Update, Delete, as well as Get and List in this case) operations. Operators use at least one controller to perform a coherent set of tasks within a cluster, usually through a combination of CRUD operations. The Operator SDK uses controller-runtime's [Client][doc-client-client] interface, which provides the interface for these operations.
+
+controller-runtime defines several interfaces used for cluster interaction:
+- `client.Client`: implementers perform CRUD operations on a Kubernetes cluster.
+- `manager.Manager`: manages shared dependencies, such as Caches and Clients.
+- `reconcile.Reconciler`: compares provided state with actual cluster state and updates the cluster on finding state differences using a Client.
+
+Clients are the focus of this document. A separate document will discuss Managers.
+
+## Client Usage
+
+### Default Client
+
+The SDK relies on a `manager.Manager` to create a `client.Client` interface that performs Create, Update, Delete, Get, and List operations within a `reconcile.Reconciler`'s Reconcile function. The SDK will generate code to create a Manager, which holds a Cache and a Client to be used in CRUD operations and communicate with the API server. By default a Controller's Reconciler will be populated with the Manager's Client which is a [split-client][doc-split-client].
+
+`pkg/controller/<kind>/<kind>_controller.go`:
+```Go
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileKind{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+type ReconcileKind struct {
+	// Populated above from a manager.Manager.
+	client client.Client
+	scheme *runtime.Scheme
+}
+```
+
+A split client reads (Get and List) from the Cache and writes (Create, Update, Delete) to the API server. Reading from the Cache significantly reduces request load on the API server; as long as the Cache is updated by the API server, read operations are eventually consistent. 
+
+### Non-default Client
+
+An operator developer may wish to create their own Client that serves read requests(Get List) from the API server instead of the cache, for example. controller-runtime provides a [constructor][doc-client-constr] for Clients:
+
+```Go
+// New returns a new Client using the provided config and Options.
+func New(config *rest.Config, options client.Options) (client.Client, error)
```

Comment:
We should probably give a small example of how to create a new client with the config. Otherwise the expectation might be the SDK provides the config(like it previously did with `sdk.GetKubeConfig()`).
```Go
import (
    "sigs.k8s.io/controller-runtime/pkg/client"
    "sigs.k8s.io/controller-runtime/pkg/client/config"
)

cfg, err := config.GetConfig()
...
c, err := client.New(cfg, client.Options{})
...
```